subworkflows/local/alignment/align_sentieon.nf

```diff
@@ -29,8 +31,20 @@ workflow ALIGN_SENTIEON {
         SENTIEON_BWAMEM.out
             .bam
             .join(SENTIEON_BWAMEM.out.bai)
-            .set { ch_bam_bai }
+            .map{ meta, bam, bai ->
+                new_meta = meta.clone()                                                                                 // clone to avoid overriding the global meta
+                new_meta.id = new_meta.id.split('_')[0]                                                                 // access the .id attribute of meta to split samplename_lane into samplename
+                new_meta.read_group = "\'@RG\\tID:" + new_meta.id + "\\tPL:" + platform + "\\tSM:" + new_meta.id + "\'"
+                [new_meta, bam, bai]}                                                                                   // end the closure to return newly modified channel
+            .groupTuple(by: 0)                                                                                          // group them bam paths with the same [ [samplename], [bam path, bam path, ..] ]
+            .branch{                                                                                                    // branch the channel into multiple channels (single, multiple) depending on size of list
+                single: it[1].size() == 1
+                multiple: it[1].size() > 1
+                }
+            .set{ merge_bams_in }
 
+        SENTIEON_READWRITER (merge_bams_in.multiple)
+        ch_bam_bai = merge_bams_in.single.mix(SENTIEON_READWRITER.out.bam_bai)
```

Comment:
Is this a channel of all bam files and also the merged bam file? I could not understand this line 😁

Reply:
This would be a channel of all bam files and bam index grouped by sample. Basically at this point each tuple will be a sample (samples with originally sequenced on one lane + samples run on multiple lanes which were then merged using readwriter algo in sentieon).